File: nmLabReport/10exponentialregression.py
# 10exponentialregression.py

import math

def exponential_regression(x, y):
    n = len(x)
    sx = sum(x)
    slgy = sum(math.log(yi) for yi in y)
    sxy = sum(x[i] * math.log(y[i]) for i in range(n))
    sx2 = sum(x[i] ** 2 for i in range(n))

    b = ((n * sxy) - (sx * slgy)) / ((n * sx2) - (sx ** 2))
    r = (slgy / n) - (b * sx / n)
    a = math.exp(r)
    return a, b

def main():
    n = int(input("Enter the number of points: "))
    x = []
    y = []
    print("Enter the value of x and fx:")
    for _ in range(n):
        xi, yi = map(float, input().split())
        x.append(xi)
        y.append(yi)

    a, b = exponential_regression(x, y)
    print(f"Fitted curve is: y = {a}e^{b}x")

if __name__ == "__main__":
    main()

File: nmLabReport/10exponentialregression.c
#include <stdio.h>
#include <conio.h>
#include <math.h>
int main()
{
    int n, i, j, k;
    float a = 0, b = 0, r, x[10], y[10], sx = 0, slgy = 0, sxy = 0, sx2 = 0;
    printf("Enter the number of points\n");
    scanf("%d", &n);
    printf("Enter the value of x and fx");
    for (i = 0; i < n; i++)
    {
        scanf("%f%f", &x[i], &y[i]);
    }
    for (i = 0; i < n; i++)
    {
        sx = sx + x[i];
        slgy = slgy + log(y[i]);
        sxy = sxy + sxy + x[i] * x[i];
    }
    b = ((n * sxy) - (sx * slgy)) / ((n * sx2) - (sx * sx));
    r = (slgy / n) - (b * sx / n);
    a = exp(r);
    printf("Fitted curve is:y= %fe ^ %fx ", a, b);
    getch();
}